src/parse_impl.rs

```diff
@@ -0,0 +1,167 @@
+use crate::parse_fn::try_consume_fn;
+use crate::parse_utils::{consume_attributes, consume_stuff_until, consume_vis_marker};
+use crate::types::{Constant, ImplMember, TyDefinition, ValueExpr};
+use crate::types_edition::GroupSpan;
+use crate::{Attribute, TyExpr, VisMarker};
+use proc_macro2::{Group, Ident, TokenTree};
+use std::iter::Peekable;
+
+type TokenIter = Peekable<proc_macro2::token_stream::IntoIter>;
+
+pub(crate) fn consume_for(tokens: &mut TokenIter) -> Option<Ident> {
+    if let Some(TokenTree::Ident(ident)) = tokens.peek() {
+        if ident == "for" {
+            let tk_for = ident.clone();
+            tokens.next();
+            return Some(tk_for);
+        }
+    }
+
+    None
+}
```

Comment:
Nitpick: this should be a `match` to be consistent with how the rest of the project does it.

Reply:
Done.